src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -386,6 +380,18 @@ public bool IsValidValue(Schema schema, GraphType type, object input)
 
             if (type is ObjectGraphType || type is InputObjectGraphType)
             {
+                if (input is KeyValuePair<string, object>)
```

Comment:
I'm not really liking this.  This type of conversion should be done on the input before the execute method is even called.  The input parser I added should make this a non-issue.  How are you passing inputs to the executor?